### PR TITLE
fix: cast_primitive data type

### PR DESCRIPTION
**Pull request by zhangweidev**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Description:
函数 cast_primitive 的 data 类型转换应该是 DVAL 

## How do you solve it?
DTVAL -> DVAL

### CODE REVIEW

**Review by wey-gu — DISMISSED** [review]
Thanks a lot for the fix/contribution!

File: .github/workflows/run_test.yaml [hunk 1]
```diff
@@ -70,12 +70,12 @@ jobs:
...
       - name: Test with pytest
         run: |
-          docker-compose -f tests/docker-compose.yaml up -d
+          docker compose -f tests/docker-compose.yaml up -d
           sleep 20
           pdm test
       - name: Test SSL connection with pytest
         run: |
-          enable_ssl=true docker-compose -f tests/docker-compose-ssl.yaml up -d
+          enable_ssl=true docker compose -f tests/docker-compose-ssl.yaml up -d
```
Comment (wey-gu) [inline .github/workflows/run_test.yaml:78]: latest gh action runner comes with a different version of docker compose, it failed here with `docker-compose`, this fixed it. ref: https://askubuntu.com/questions/1508129/docker-compose-giving-containerconfig-errors-after-update-today

**Comment by wey-gu** [comment]
Need to look into this ci error.

```
=================================== FAILURES ===================================
________________________ test_remove_invalid_connection ________________________

    def test_remove_invalid_connection():
        addresses = [("127.0.0.1", 9669), ("127.0.0.1", 9670), ("127.0.0.1", 9671)]
        configs = Config()
        configs.min_connection_pool_size = 30
        configs.max_connection_pool_size = 45
        pool = ConnectionPool()
    
        try:
            assert pool.init(addresses, configs)
    
            # turn down one server('127.0.0.1', 9669) so the connection to it is invalid
            os.system("docker stop tests_graphd0_1")
            time.sleep(3)
    
            # get connection from the pool, we should be able to still get 30 connections even though one server is down
            for i in range(0, 30):
                conn = pool.get_connection()
                assert conn is not None
    
            # total connection should still be 30
            assert pool.connects() == 30
    
            # the number of connections to the down server should be 0
>           assert len(pool._connections[addresses[0]]) == 0
E           assert 10 == 0
E            +  where 10 = len(deque([<nebula3.gclient.net.Connection.Connection object at 0x7f4debc84700>, <nebula3.gclient.net.Connection.Connection object at 0x7f4dec88cf10>, <nebula3.gclient.net.Connection.Connection object at 0x7f4dec880a30>, <nebula3.gclient.net.Connection.Connection object at 0x7f4decd3c190>, <nebula3.gclient.net.Connection.Connection object at 0x7f4decd54df0>, <nebula3.gclient.net.Connection.Connection object at 0x7f4decd54400>, ...]))

tests/test_pool.py:298: AssertionError
```

**Review by wey-gu — DISMISSED** [review]

File: tests/test_pool.py [hunk 2]
```diff
@@ -283,7 +283,7 @@ def test_remove_invalid_connection():
         assert pool.init(addresses, configs)
 
         # turn down one server('127.0.0.1', 9669) so the connection to it is invalid
-        os.system("docker stop tests_graphd0_1")
+        os.system("docker stop tests_graphd0_1 || docker stop tests-graphd0-1")
```
Comment (wey-gu) [inline tests/test_pool.py:286]: The new docker-compose comes with a different naming per container...

**Review by wey-gu — APPROVED** [review]

**Comment by wey-gu** [comment]
@Nicole00 

Only one line of change, other changes are due to github action runner change, caused docker-compose broken in Ubuntu 22.04.

**Review by BeautyyuYanli — APPROVED** [review]

**Review by Nicole00 — APPROVED** [review]

**Pull request merged**